VAMobile/src/screens/HealthScreen/Appointments/AppointmentDetailsCommon/PendingAppointmentCancelButton.tsx

```diff
@@ -26,11 +29,26 @@ const PendingAppointmentCancelButton: FC<PendingAppointmentCancelButtonProps> =
   const { cancelId, status } = attributes || ({} as AppointmentAttributes)
 
   if (isAppointmentPending && cancelId && status !== AppointmentStatusConstants.CANCELLED) {
+    let apiStatus: string | undefined
+    const isPendingAppointment = attributes.isPending && (attributes.status === AppointmentStatusConstants.SUBMITTED || attributes.status === AppointmentStatusConstants.CANCELLED)
+    if (attributes.status === AppointmentStatusConstants.CANCELLED) {
+      apiStatus = 'Canceled'
+    } else if (attributes.status === AppointmentStatusConstants.BOOKED) {
+      apiStatus = 'Confirmed'
+    } else if (isPendingAppointment) {
+      apiStatus = 'Pending'
+    }
+    const apptDate = Math.floor(DateTime.fromISO(attributes.startDateUtc).toMillis() / (1000 * 60 * 60 * 24))
+    const nowDate = Math.floor(DateTime.now().toMillis() / (1000 * 60 * 60 * 24))
+    const days = apptDate - nowDate
```

Comment:
This logic is used in a few files, can this be refactored into a function and stored in the appointments or analytics util file?

Reply:
agreed, pulled out to separate functions for that logic